lib/payex/api/pxorder.rb

```diff
@@ -103,65 +103,83 @@ def PurchaseInvoicePrivate(params)
         signed: true
       },
       'customerRef' => {
-        signed: true
+        signed: true,
+        format: /[a-zA-Z0-9]{,15}/
       },
       'customerName' => {
-        signed: true
+        signed: true,
+        format: /^[a-zA-Z0-9_:!;\\”#<>=?\\[\\]@{}´\n\r %-\/À-ÖØ-öø-úü]{,35}$?/,
+        resolve: proc { |v| v[0..35].strip }
       },
       'streetAddress' => {
-        signed: true
+        signed: true,
+        format: /^[a-zA-Z0-9_:!;\\”#<>=?\\[\\]@{}´\n\r %-\/À-ÖØ-öø-úü]{,35}$?/
       },
       'coAddress' => {
-        signed: true
+        signed: true,
+        format: /^[a-zA-Z0-9_:!;\\”#<>=?\\[\\]@{}´\n\r %-\/À-ÖØ-öø-úü]{,35}$?/
       },
       'postalCode' => {
-        signed: true
+        signed: true,
+        format: /[A-Z0-9\-]{,9}/
       },
       'city' => {
-        signed: true
+        signed: true,
+        format: /^[a-zA-Z0-9_:!;\\”#<>=?\\[\\]@{}´\n\r %-\/À-ÖØ-öø-úü]{,27}$?/
       },
       'country' => {
-        signed: true
+        signed: true,
+        format: /[A-Z]{,2}$/
       },
       'socialSecurityNumber' => {
-        signed: true
+        signed: true,
+        format: /^[a-zA-Z0-9_:!;\\”#<>=?\\[\\]@{}´\n\r %-\/À-ÖØ-öø-úü]{,15}$?/
```

Comment:
I'm not entirely sure this regex is correct. Try this instead `\A[a-zA-Z0-9_:!;\"#<>=?\\[\\]@{}´\n\r %-\/À-ÖØ-öø-úü]*\z`.